python/hail/plot/plots.py

```diff
@@ -133,3 +231,98 @@ def qq(pvals):
     bound = max(max(exp), max(obs)) * 1.1
     p.line([0, bound], [0, bound], color='red')
     return p
+
+
+@typecheck(pvals=expr_float64, locus=nullable(expr_locus()), title=nullable(str),
+           size=int, hover_fields=nullable(dictof(str, sequenceof(anytype))))
+def manhattan(pvals, locus=None, title=None, size=4, hover_fields=None):
+    """Create a Manhattan plot. (https://en.wikipedia.org/wiki/Manhattan_plot)
+
+    Parameters
+    ----------
+    pvals : :class:`.Float64Expression`
+        P-values to be plotted.
+    locus : :class:`.LocusExpression`
+        Locus values to be plotted.
+    title : str
+        Title of the plot.
+    size : int
+        Size of markers in screen space units.
+    hover_fields: Dict[str, List[Any]]
+        Dictionary of field names and values to be shown in the HoverTool of the plot.
+
+    Returns
+    -------
+    :class:`bokeh.plotting.figure.Figure`
+    """
+    def get_contig_index(x, starts):
+        left = 0
+        right = len(starts) - 1
+        while left <= right:
+            mid = int((left + right) / 2)
+            if x < starts[mid]:
+                if x >= starts[mid - 1]:
+                    return mid - 1
+                right = mid
+            elif x >= starts[mid+1]:
+                left = mid + 1
+            else:
+                return mid
+
+    pvals = -hail.log10(pvals)
+
+    if locus is None:
+        locus = pvals._indices.source.locus
+    res = hail.tuple([locus.global_position(), pvals, locus.contig, locus.position]).collect()
+    x = [point[0] for point in res]
+    y = [point[1] for point in res]
+    label = [point[2] for point in res]
+    variant_identifier = [str(point[2]) + ':' + str(point[3]) for point in res]
+
+    if hover_fields is None:
+        hover_fields = {'variant_identifier': variant_identifier}
+    else:
+        hover_fields['variant_identifier'] = variant_identifier
```

Comment:
I think this piece is broken -- if I pass `hover_fields = {'gene': mt.gene}` we never collect the gene, it looks like

Reply:
I think you'll need to put the hover fields logic above the `collect`, and collect the label fields as a `hl.struct` there

Reply:
does that make sense?